tests/test_all_rules.py

```diff
@@ -814,3 +815,27 @@ def test_note_guide(self):
                 if osquery_note_pattern not in rule.contents.data.note:
                     self.fail(f'{self.rule_str(rule)} Investigation guides using the Osquery Markdown must contain '
                               f'the following note:\n{osquery_note_pattern}')
+
+
+class TestEndpointQuery(BaseRuleTest):
+    """Test endpoint-specific rules."""
+
+    def test_os_and_platform_in_query(self):
+        """Test that all endpoint rules have an os defined and linux includes platform."""
+        for rule in self.production_rules:
+            if not rule.contents.data.get('language') in ('eql', 'kuery'):
+                continue
+            if rule.path.parent.name not in ('windows', 'macos', 'linux'):
+                # skip cross-platform for now
+                continue
+
+            ast = rule.contents.data.ast
+            fields = [str(f) for f in ast if isinstance(f, (kql.ast.Field, eql.ast.Field))]
+
+            err_msg = f'{self.rule_str(rule)} missing required field for endpoint rule'
+            self.assertIn('host.os.type', fields, err_msg)
+
+            # going to bypass this for now
+            # if rule.path.parent.name == 'linux':
+            #     err_msg = f'{self.rule_str(rule)} missing required field for linux endpoint rule'
+            #     self.assertIn('host.os.platform', fields, err_msg)
```

Comment:
One item of the checklist in the issue for this PR is:

` Linux should have target distributions too (need to verify the most consistent field)` and  `also should unit test this`. Is this task item relevant to this section of code?

Reply:
ywp, distribution == platform